### PR TITLE
Use Docc for documentation

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation

Documentation is nice, and we can help support users by providing useful
clear docs.

Modifications

Add Docc to 5.6 and later builds
Make sure symbol references work
Add overview docs

Result

Nice rendering docs

### CODE REVIEW

File: Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift [hunk 1]
```diff
@@ -16,12 +16,21 @@
 import NIOCore
 import NIOHTTP1
 
+/// A representation of a HTTP request for the Swift Concurrency HTTPClient API.
+///
+/// This object is similar to ``HTTPClient/Request``, but used for the Swift Concurrency API.
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
 public struct HTTPClientRequest {
+    /// The request URL, including scheme and hostname.
```
Comment (FranzBusch) [inline Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift:24]: Should this also include the port?

File: Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift [hunk 2]
```diff
@@ -34,6 +43,10 @@ public struct HTTPClientRequest {
 
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
 extension HTTPClientRequest {
+    /// A HTTP request body.
```
Comment (FranzBusch) [inline Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift:46]:
```suggestion
    /// An HTTP request body.
```

File: Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift [hunk 3]
```diff
@@ -54,10 +67,20 @@ extension HTTPClientRequest {
 
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
 extension HTTPClientRequest.Body {
+    /// Create a ``HTTPClientRequest/Body-swift.struct`` from a `ByteBuffer`.
```
Comment (FranzBusch) [inline Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift:70]:
```suggestion
    /// Create an ``HTTPClientRequest/Body-swift.struct`` from a `ByteBuffer`.
```

File: Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift [hunk 4]
```diff
@@ -54,10 +67,20 @@ extension HTTPClientRequest {
...
+    ///
+    /// - parameter byteBuffer: The bytes of the body.
     public static func bytes(_ byteBuffer: ByteBuffer) -> Self {
         self.init(.byteBuffer(byteBuffer))
     }
 
+    /// Create a ``HTTPClientRequest/Body-swift.struct`` from a `RandomAccessCollection` of bytes.
```
Comment (FranzBusch) [inline Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift:77]:
```suggestion
    /// Create an ``HTTPClientRequest/Body-swift.struct`` from a `RandomAccessCollection` of bytes.
```

File: Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift [hunk 5]
```diff
@@ -75,6 +98,23 @@ extension HTTPClientRequest.Body {
         })
     }
 
+    /// Create a ``HTTPClientRequest/Body-swift.struct`` from a `Sequence` of bytes.
```
Comment (FranzBusch) [inline Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift:101]:
```suggestion
    /// Create an ``HTTPClientRequest/Body-swift.struct`` from a `Sequence` of bytes.
```

File: Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift [hunk 6]
```diff
@@ -111,6 +164,17 @@ extension HTTPClientRequest.Body {
         })
     }
 
+    /// Create a ``HTTPClientRequest/Body-swift.struct`` from an `AsyncSequence` of `ByteBuffer`s.
```
Comment (FranzBusch) [inline Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift:167]:
```suggestion
    /// Create an ``HTTPClientRequest/Body-swift.struct`` from an `AsyncSequence` of `ByteBuffer`s.
```

File: Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift [hunk 7]
```diff
@@ -123,6 +187,18 @@ extension HTTPClientRequest.Body {
         return body
     }
 
+    /// Create a ``HTTPClientRequest/Body-swift.struct`` from an `AsyncSequence` of bytes.
+    ///
+    /// This construction will consume 1kB chunks from the `Bytes` and send them at once. This optimizes for
+    /// `AsyncSEquence`s where
```
[the remark this answers is not in the document]
Comment (FranzBusch) [reply Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift:193]: You stopped mid sentence here

File: Sources/AsyncHTTPClient/AsyncAwait/HTTPClientResponse.swift [hunk 8]
```diff
@@ -16,13 +16,28 @@
...
+    /// The HTTP status for this response.
     public var status: HTTPResponseStatus
+
+    /// The HTTP headers of this response.
     public var headers: HTTPHeaders
+
+    /// The body of this HTTP response.
     public var body: Body
 
+    /// A representation of the response body for a HTTP response.
```
Comment (FranzBusch) [inline Sources/AsyncHTTPClient/AsyncAwait/HTTPClientResponse.swift:36]:
```suggestion
    /// A representation of the response body for an HTTP response.
```

**Review by FranzBusch — APPROVED** [review]
Looks great! Just a bunch of nits

**Review by dnadoba — APPROVED** [review]

**Pull request merged**